v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerLT100Gb.java

```diff
@@ -0,0 +1,197 @@
+            TEST_ROOT_DIR + testName + "dlq",
+            pubsubResourceManager,
+            dlqGcsPrefix,
+            gcsResourceManager);
+
+    // Setup Datastream
+    MySQLSource mySQLSource = getMySQLSource();
+    Stream stream =
+        createDatastreamResources(
+            ARTIFACT_BUCKET, gcsPrefix, mySQLSource, datastreamResourceManager);
+
+    // Setup Parameters
+    Map<String, String> params =
+        new HashMap<>() {
+          {
+            put("inputFilePattern", getGcsPath(ARTIFACT_BUCKET, gcsPrefix));
+            put("streamName", stream.getName());
+            put("instanceId", spannerResourceManager.getInstanceId());
+            put("databaseId", spannerResourceManager.getDatabaseId());
+            put("projectId", project);
+            put("deadLetterQueueDirectory", getGcsPath(ARTIFACT_BUCKET, dlqGcsPrefix));
+            put("gcsPubSubSubscription", subscription.toString());
+            put("dlqGcsPubSubSubscription", dlqSubscription.toString());
+            put("datastreamSourceType", "mysql");
+            put("inputFileFormat", "avro");
+          }
+        };
+
+    LaunchConfig.Builder options = LaunchConfig.builder(getClass().getSimpleName(), SPEC_PATH);
+    options.addEnvironment("maxWorkers", MAX_WORKERS).addEnvironment("numWorkers", NUM_WORKERS);
+
+    options.setParameters(params);
+
+    // Act
+    PipelineLauncher.LaunchInfo jobInfo = pipelineLauncher.launch(project, region, options.build());
+    assertThatPipeline(jobInfo).isRunning();
+
+    ConditionCheck[] checks = new ConditionCheck[10];
+    for (int i = 0; i < 10; ++i) {
+      checks[i] =
+          SpannerRowsCheck.builder(spannerResourceManager, TABLE + (i + 1))
+              .setMinRows(6500000)
+              .setMaxRows(6500000)
+              .build();
+    }
+
+    PipelineOperator.Result result =
+        pipelineOperator.waitForCondition(
+            createConfig(jobInfo, Duration.ofHours(4), Duration.ofMinutes(5)), checks);
+
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+
+    result = pipelineOperator.cancelJobAndFinish(createConfig(jobInfo, Duration.ofMinutes(5)));
+    assertThatResult(result).isLaunchFinished();
+
+    // export results
+    exportMetricsToBigQuery(jobInfo, getMetrics(jobInfo));
+  }
+
+  public MySQLSource getMySQLSource() {
+    String hostIp =
+        secretClient.accessSecret(
+            "projects/269744978479/secrets/nokill-datastream-mysql-to-spanner-cloudsql-ip-address/versions/1");
+    String username =
+        secretClient.accessSecret(
+            "projects/269744978479/secrets/nokill-datastream-mysql-to-spanner-cloudsql-username/versions/1");
+    String password =
+        secretClient.accessSecret(
+            "projects/269744978479/secrets/nokill-datastream-mysql-to-spanner-cloudsql-password/versions/1");
+    MySQLSource mySQLSource = new MySQLSource.Builder(hostIp, username, password, 3306).build();
+    return mySQLSource;
```

Comment:
Is this how it's done in other places as well?

Reply:
We have directly written ip address in other places. I think this is better way.